pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -681,11 +690,23 @@ protected SegmentBuildDescriptor buildSegmentInternal(boolean forCommit) {
           TimeUnit.MILLISECONDS.toSeconds(waitTimeMillis));
 
       if (forCommit) {
+        Preconditions.checkState(destDir.listFiles() != null && destDir.listFiles().length > 0,
+            "The index dir is empty: ", destDir);
+        File metadataFileName = new File(destDir.listFiles()[0], V1Constants.MetadataKeys.METADATA_FILE_NAME);
```

Comment:
destDir.listFiles() is being called 4 times. Might be cleaner if we store it in a local variable

Reply:
Done.